ocs_ci/resiliency/platform_stress.py

```diff
@@ -0,0 +1,298 @@
+            vm (int, optional): Number of VM stressors. Defaults to 1.
+            io (int, optional): Number of I/O stressors. Defaults to 4.
+            timeout (int, optional): Timeout in seconds. Defaults to 60.
+
+        Returns:
+            bool: True if stress command succeeded, False otherwise.
+        """
+        if self.stop_event.is_set():
+            return False
+        cpu_count = int(node_obj.data["status"]["capacity"]["cpu"])
+        sock = cpu_count
+        log.info(
+            "Running combined stress test: CPU=%d, VM=%d, IO=%d, SOCK=%d",
+            cpu_count,
+            vm,
+            io,
+            sock,
+        )
+        return self._apply_stress(
+            node_obj,
+            f" --cpu {cpu_count} --vm {vm} --iomix {io} --sock {sock} --temp-path /mnt/disk1 ",
+            timeout,
+        )
+
+    def _run_random_stress_loop(self, timeout=0, node_selection="ALL"):
+        """Internal method that runs in a loop to apply random stress tests.
+
+        Args:
+            timeout (int, optional): Timeout for each stress test. 0 means random.
+            node_selection (str, optional): "ALL" for all nodes or "RANDOM" for subset.
+        """
+        stress_funcs = [
+            self.cpu_stress,
+            self.memory_stress,
+            self.io_stress,
+            self.network_stress,
+            self.all_stress,
+        ]
+
+        while not self.stop_event.is_set():
+            if not self.nodes:
+                log.warning("No nodes available for stress test.")
+                break
+
+            subset_nodes = (
+                self.nodes
+                if node_selection.upper() == "ALL"
+                else random.sample(self.nodes, min(2, len(self.nodes)))
+            )
+            current_timeout = timeout or random.randint(30, 120)
+            selected = random.choice(stress_funcs)
+            log.info(
+                "Randomly selected %s for nodes: %s",
+                selected.__name__,
+                [n.name for n in subset_nodes],
+            )
+
+            for node in subset_nodes:
+                if self.stop_event.is_set():
+                    break
+                thread = threading.Thread(
+                    target=selected, args=(node,), kwargs={"timeout": current_timeout}
+                )
+                thread.start()
+                self.active_threads.append(thread)
+                log.info(
+                    "Started thread for %s on node: %s", selected.__name__, node.name
+                )
+
+            # Wait for current stress tests to complete or stop event
+            for thread in list(self.active_threads):
+                if self.stop_event.is_set():
+                    break
+                thread.join(timeout=1)
+                if not thread.is_alive():
+                    self.active_threads.remove(thread)
+
+            log.info("Cooling down for 30 seconds before next stress test...")
```

Comment:
if possible, let's avoid sleep and use some health validation

Reply:
Added ceph health check call